geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -622,7 +623,9 @@ public void computeWithPrimaryLocked(Object key, Runnable r) throws PrimaryBucke
     }
 
     try {
-      br.doLockForPrimary(false);
+      if (!br.doLockForPrimary(false)) {
```

Comment:
remove the "if" and just call br.doLockForPrimary(false);

Reply:
just updated it!